include/dump_offload.hpp

```diff
@@ -0,0 +1,443 @@
+                if (ec == boost::system::errc::no_such_file_or_directory ||
+                    ec == boost::system::errc::connection_refused)
+                {
+                    BMCWEB_LOG_DEBUG("UNIX Socket: async_connect {} {}",
+                                     ec.message(), ec);
+                    retrySocketConnect();
+                    return;
+                }
+                BMCWEB_LOG_ERROR("UNIX Socket: async_connect error {} {}",
+                                 ec.message(), ec);
+                waitTimer.cancel();
+                this->connection->sendStreamErrorStatus(
+                    boost::beast::http::status::internal_server_error);
+                this->connection->close();
+                this->cleanupSocketFiles();
+                return;
+            }
+            waitTimer.cancel();
+            this->connection->sendStreamHeaders(std::to_string(this->dumpSize),
+                                                "application/octet-stream");
+            this->doReadStream();
+        });
+    }
+
+    /**
+     * @brief  Invokes InitiateOffload method of dump manager which
+     *         directs dump manager to start writing on unix domain socket.
+     *
+     * @return void
+     */
+    void initiateOffload()
+    {
+        crow::connections::systemBus->async_method_call(
+            [this,
+             self(shared_from_this())](const boost::system::error_code ec) {
+            if (ec)
+            {
+                BMCWEB_LOG_ERROR("DBUS response error: {}", ec);
+                if (ec.value() == EBADR)
+                {
+                    this->connection->sendStreamErrorStatus(
+                        boost::beast::http::status::not_found);
+                }
+                else
+                {
+                    this->connection->sendStreamErrorStatus(
+                        boost::beast::http::status::internal_server_error);
+                }
+                this->connection->close();
+                this->cleanupSocketFiles();
+                return;
+            }
+        },
+            "xyz.openbmc_project.Dump.Manager",
+            "/xyz/openbmc_project/dump/" + dumpType + "/entry/" + entryID,
+            "xyz.openbmc_project.Dump.Entry", "InitiateOffload",
+            unixSocketPath.c_str());
+    }
+
+    /**
+     * @brief  This function setup a timer for retrying unix socket connect.
+     *
+     * @return void
+     */
+    void retrySocketConnect()
+    {
+        waitTimer.expires_after(std::chrono::milliseconds(1000));
+
+        waitTimer.async_wait([this, self(shared_from_this())](
+                                 const boost::system::error_code& ec) {
+            if (ec)
+            {
+                BMCWEB_LOG_ERROR("Async_wait failed {}", ec);
+                return;
+            }
+
+            if (connectRetryCount < maxConnectRetryCount)
+            {
+                BMCWEB_LOG_ERROR("Failed to connect, reached max retry count: ",
```

Comment:
Do we really need to log this?

Reply:
yeah, we need all these in failure cases